README.md

```diff
@@ -103,13 +103,50 @@ data:
 
 #### Set up Cloud Backup With GCP Secret Manager
 
-The `secret-agent` expects credentials to be discoverable via standard [GCP mechanisms](https://cloud.google.com/docs/authentication). These credentials can be provided in a number of ways, including user accounts and service accounts.
+The `secret-agent` expects credentials to be discoverable via standard [GCP mechanisms](https://cloud.google.com/docs/authentication). These credentials can be provided in a number of ways, including:
+
+* [Workload identity](#workload-identity) (recommended for GKE deployments)
+* [Credentials file](#credentials-file) with user accounts or service accounts.
 
 Please refer to the [GCP Documentation](https://cloud.google.com/secret-manager/docs/reference/libraries?hl=nl#cloud-console) for instructions on how to create a service account with the necessary permissions to access the GCP Secrets Manager. The `secret-agent` needs access to read/write secrets. This can be achieved by assigning the `Secret Manager Admin` role to the service account provided.
 
+##### Workload identity
+
+Workload Identity is the recommended way to access Google Cloud services from applications running within GKE. For more information on how to enable workload identity see [GCP Documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#gcloud_1).
+
+In general, once the Google service account has been created with the proper role attached and workload identity has been enabled for your GKE cluster and nodes, you can run the following commands to enable workload identity for the `secret-agent`:
+
+```bash
```

Comment:
I question if we should offer these instructions other than a pointer to the documents

Reply:
I guess we could. Make the `cloud-credentials` reference optional and remove the references to `AZURE_MANAGED_ID`. Instead of setting that to "true", just skip the credentials file altogether. If no reference is provided, do nothing (like what currently happens if `AZURE_MANAGED_ID` is set to "true")